lib/os/onoff.c

```diff
@@ -478,10 +490,16 @@ int onoff_service_reset(struct onoff_service *srv,
 	return rv;
 }
 
+static void onoff_cancel_notify(struct onoff_service *srv,  int res)
+{
+	onoff_stop_notify(srv, res >= 0 ? -ECANCELED : res);
```

Comment:
at this point successful cancellation is like stop but with specific error code.